src/core/config.py

```diff
@@ -0,0 +1,71 @@
+from typing import Annotated, Any
+
+from dotenv import find_dotenv
+from pydantic import BeforeValidator, HttpUrl, SecretStr, TypeAdapter, computed_field
+from pydantic_settings import BaseSettings, SettingsConfigDict
+
+from core.llm import AnthropicModelName, GoogleModelName, GroqModelName, OpenAIModelName
+
+
+def check_str_is_http(x: str) -> str:
+    http_url_adapter = TypeAdapter(HttpUrl)
+    return str(http_url_adapter.validate_python(x))
+
+
+class Settings(BaseSettings):
+    model_config = SettingsConfigDict(
+        env_file=find_dotenv(),
+        env_file_encoding="utf-8",
+        env_ignore_empty=True,
+        extra="ignore",
+    )
+    MODE: str | None = None
+
+    HOST: str = "0.0.0.0"
+    PORT: int = 80
+
+    AUTH_SECRET: SecretStr | None = None
+
+    OPENAI_API_KEY: SecretStr | None = None
+    ANTHROPIC_API_KEY: SecretStr | None = None
+    GOOGLE_API_KEY: SecretStr | None = None
+    GROQ_API_KEY: SecretStr | None = None
+
+    OPENAI_DEFAULT_MODEL: OpenAIModelName = OpenAIModelName.GPT_4O_MINI
+    ANTHROPIC_DEFAULT_MODEL: AnthropicModelName = AnthropicModelName.HAIKU_3
+    GOOGLE_DEFAULT_MODEL: GoogleModelName = GoogleModelName.GEMINI_15_FLASH
+    GROQ_DEFAULT_MODEL: GroqModelName = GroqModelName.LLAMA_31_70B
```

Comment:
Is the idea here that the developer could change the default model by setting this environment variable?

I think it might be overkill, and a bit neater to just have one `DEFAULT_MODEL = OpenAIModelName.GPT_40_MINI` defined in `core.llm` instead.

Then the model creation in agents would be simpler:

```python
from core.llm import get_model, DEFAULT_MODEL
# ...
get_model(config["configurable"].get("model", DEFAULT_MODEL))
```

Reply:
Alternatively, if you want to make this configurable, can we instead do a simpler `settings.DEFAULT_MODEL` ? And you could check that it's valid for the API key, and if it isn't set in `model_post_init()` you could check which API keys have been set and pick a default model based on that.

OK, I talked myself into maybe it's a separate PR - if you want to defer I can file an issue

Reply:
fixed